### PR TITLE
Improve csrf protection take2

### PR DESCRIPTION
**Pull request by rmescandon**
fixes bug #57

 - Updated csrf token before any put, post or delete operation by calling explicit /v1/csrf api resource. Got token from response and injected as new header
 - Bump version to 1.5

### CODE REVIEW

File: service/router.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func AdminRouter(env *Env) *mux.Router {
 	// Start the web service router
 	router := mux.NewRouter()
 
+	// API routes: csrf token
+	router.Handle("/v1/csrf", Middleware(http.HandlerFunc(TokenHandler), env)).Methods("GET")
```
Comment (slimjim777) [inline service/router.go]: Can we call this /v1/token instead? I'm wondering if it should respond to PUT to make it a little more secure.

File: webapp/js/models/Ajax.js [hunk 2]
```diff
@@ -15,45 +15,59 @@
...
  */
 'use strict'
-var request =require('then-request');
+var request = require('then-request');
 var API_VERSION = '/v1/';
 
 var Ajax = {
+
+	getToken: function() {
+		return this.get('csrf')
```
Comment (slimjim777) [inline webapp/js/models/Ajax.js]: This endpoint would need to get renamed as well i.e. csrf => token

File: service/router.go (hunk 1 above)
Comment (rmescandon) [inline service/router.go]: sounds to me quite generic having "token" in the path.. token of what? a nonce is also a token. makes sense /v1/csrftoken ?

Comment (slimjim777) [inline service/router.go]: That was the idea - keep it generic, rather than advertising the endpoint as a csrftoken.

**Pull request merged**